### DISCUSSION TITLE
Separate encoding server from web frontend

### DISCUSSION THREAD

**Discussion by fluffy-critter** [owncast/owncast#3872] [Ideas] [answered]
It would be really nice if I could separate out the part of Owncast that handles the web frontend from the streaming server.

For example, I have a gigabit connection at home and could throw physical encoding hardware at it, rather than being limited to software encoding on my Linode instance, but I don't want to always have the Owncast instance running from my home network when I'm not actively streaming.

Similarly, it would be nice to be able to always have my Owncast web frontend available on a shared server or the like, but then only spin up a higher-performance cloud encoder instance on an as-needed basis and take advantage of the hourly rates for such things.

Ideally this could also be combined with using the S3-style object storage, which would enable people to do multiple high-quality encoding streams from their physical hardware at home without needing to worry about their home connection becoming saturated if they get a lot of viewers.

**Comment by gabek** [comment, marked as answer]
I understand the value of this for certain use cases, but this just isn't the solution that we're offering. That's something different from Owncast.

**Reply by fluffy-critter** [reply]
Thanks, what I've ended up doing is I'm running owncast from my home network and then running a caching nginx proxy on the public web and that's working pretty well for me.
